File: 026/main.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
  if (nums.length === 0) {
    return 0
  }
  let point = 1
  for (let i = 1; i < nums.length; i++) {
    if (nums[i] !== nums[i - 1]) {
      nums[point++] = nums[i]
    }
  }
  nums.splice(point)
  return nums.length
}

let nums = [1, 2, 2, 3, 3]
console.log(removeDuplicates(nums))
console.log(nums)

File: 026/main.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        prev, slow = float('-inf'), 0
        for i, item in enumerate(nums):
            if item != prev:
                nums[slow] = item
                slow += 1
            prev = item
        return slow

import os

if os.getenv('LZS'):
    s = Solution()
    lst = [1, 1, 2, 2, 3, 4, 5, 5]
    print(s.removeDuplicates(lst))
    print(lst)
